mhkit/wave/contours.py

```diff
@@ -1744,34 +1744,55 @@ def samples_contour(t_samples, t_contour, hs_contour):
     hs_samples : nparray
         points sampled along return contour
     """
-    assert isinstance(
-        t_samples, np.ndarray), 't_samples must be of type np.ndarray'
-    assert isinstance(
-        t_contour, np.ndarray), 't_contour must be of type np.ndarray'
-    assert isinstance(
-        hs_contour, np.ndarray), 'hs_contour must be of type np.ndarray'
-
-    # finds minimum and maximum energy period values
+    if not isinstance(t_samples, np.ndarray):
+        raise TypeError('t_samples must be of type np.ndarray')
+    if not isinstance(t_contour, np.ndarray):
+        raise TypeError('t_contour must be of type np.ndarray')
+    if not isinstance(hs_contour, np.ndarray):
+        raise TypeError('hs_contour must be of type np.ndarray')
+    if len(t_contour) != len(hs_contour):
+        raise ValueError(
+            "t_contour and hs_contour must be of the same length.")
+    if np.any(t_samples < np.min(t_contour)) or np.any(t_samples > np.max(t_contour)):
+        raise ValueError(
+            "All t_samples must be within the range of t_contour.")
+
+    # Find minimum and maximum energy period values
     amin = np.argmin(t_contour)
     amax = np.argmax(t_contour)
     aamin = np.min([amin, amax])
     aamax = np.max([amin, amax])
-    # finds points along the contour
+
+    # Separate points along the contour into upper & lower half
     w1 = hs_contour[aamin:aamax]
     w2 = np.concatenate((hs_contour[aamax:], hs_contour[:aamin]))
-    if (np.max(w1) > np.max(w2)):
-        x1 = t_contour[aamin:aamax]
-        y1 = hs_contour[aamin:aamax]
+
+    # Get samples min and max
+    t_min, t_max = np.min(t_samples), np.max(t_samples)
+
+    # Choose the half of the contour with the largest wave height
+    if np.max(w1) > np.max(w2):
+        # Check if the max or min Tp values are within the contour half
+        include_aamax = t_max >= t_contour[aamax] or t_min <= t_contour[aamin]
+        # Set the x and y values for interpolation
+        x1 = t_contour[aamin:aamax + int(include_aamax)]
+        y1 = hs_contour[aamin:aamax + int(include_aamax)]
     else:
-        x1 = np.concatenate((t_contour[aamax:], t_contour[:aamin]))
-        y1 = np.concatenate((hs_contour[aamax:], hs_contour[:aamin]))
-    # sorts data based on the max and min energy period values
+        # Check if the max or min Tp values are within the contour half
+        include_aamin = t_max >= t_contour[aamin] or t_min <= t_contour[aamin]
+        # Set the x and y values for interpolation
+        x1 = np.concatenate(
+            (t_contour[aamax:], t_contour[:aamin + int(include_aamin)]))
+        y1 = np.concatenate(
+            (hs_contour[aamax:], hs_contour[:aamin + int(include_aamin)]))
```

Comment:
Same reasoning as above, 
```suggestion
        x1 = np.concatenate(
            (t_contour[aamax - int(include_aamin):], t_contour[:aamin + int(include_aamin)]))
        y1 = np.concatenate(
            (hs_contour[aamax - int(include_aamin):], hs_contour[:aamin + int(include_aamin)]))